o1vm/src/mips/constraints.rs

```diff
@@ -481,6 +482,31 @@ impl<Fp: Field> InterpreterEnv for Env<Fp> {
                 );
             }
 
+            // Constrain that at most you read `len` bytes
```

Comment:
Don't we simply want to check that `actual_read_len <= len`, i.e `0 <= len - actual_read_len <= 4`. Can be done using a lookup.

Reply:
Let's move the conversation in https://github.com/o1-labs/proof-systems/issues/2335.

Reply:
That's a good thing to check, apart from distinguishing between bytelength bytes and preimage bytes. Because only the preimage bytes must be part of the absorb lookup. So the reason for having chunks is to prove decomposition of preimage bytes as 4byte chunks. 

But yeah, I like your approach if we don't mind adding a small table for numbers 0-4.